src/core/symbology-ng/qgssymbollayerutils.cpp

```diff
@@ -3986,8 +4003,26 @@ double QgsSymbolLayerUtils::rescaleUom( double size, QgsUnitTypes::RenderUnit un
           scale = 1 / 0.28;
           roundToUnit = true;
           break;
-        // we don't have a good case for map units, as pixel values won't change based on zoom
-        default:
+        case QgsUnitTypes::RenderInches:
+          scale = 1 / 0.28 * 25.4;
+          roundToUnit = true;
+          break;
+        case QgsUnitTypes::RenderPoints:
+          scale = 90. /* dots per inch according to OGC SLD */ / 72. /* points per inch */;
+          roundToUnit = true;
+          break;
+        case QgsUnitTypes::RenderPixels:
+          // pixel is pixel
+          scale = 1;
+          break;
+        case QgsUnitTypes::RenderMapUnits:
+        case QgsUnitTypes::RenderMetersInMapUnits:
+          // already handed via uom
+          scale = 1;
+          break;
+        case QgsUnitTypes::RenderPercentage:
+        case QgsUnitTypes::RenderUnknownUnit:
+          // these do not make sense and should not really reach here
```

Comment:
@luipir Actually I don't think we should *ever* be using default: switches with enums (with the one exception of switching over QVariant types or WKB types). It's much safer to explicitly list all enum values, and leave off the default so that a warning is thrown. This means that you're always aware of ALL the code which needs to be handled when adding new values to an existing enum.